From 688c1df5ed4b66b369d8e39ebc59b7a9b8ff0310 Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Sat, 9 Feb 2019 10:30:23 +0530
Subject: [PATCH 0074/1921] octeontx2-af: Skip creating limits pci file for
 undiscovered RVU PFs

If not all of 15 RVU PFs (except AF) are enumerated in kernel then
current logic results in a kernel crash. This patch fixes it by
skipping 'limits' pci file for undiscovered PFs.

Change-Id: I84e8cfa600bb1dfdebbb388ac2bd4bc403fce6e7
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_validation.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_validation.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_validation.c
index 0128eaa070ff..04573e110d4e 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_validation.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_validation.c
@@ -525,6 +525,8 @@ static int rvu_create_limits_sysfs(struct rvu *rvu)
 
 	for (i = 0; i < rvu->hw->total_pfs; i++) {
 		pf = &rvu->pf[i];
+		if (!pf->pdev)
+			continue;
 		pdev = pf->pdev;
 
 		pf->limits_kobj = kobject_create_and_add("limits",
-- 
2.31.1

